### ISSUE TITLE
identifier quote check is incorrect when passing mixed-case names

### ISSUE THREAD

**Issue by urkle** [tlocke/pg8000#157]
the identifier function checks to see if the passed identifier contains non alpha and non-numeric contents.  However it should be checking the case of the alpha characters as well.

```python
# what it does right now
identifier('myAwesomeTable') == 'myAwesomeTable'
# what it should do
identifier('myAwesomeTable') == '"myAwesomeTable"'
```

Otherwise Postgres case-folds the identifier (myAwesomeTable -> myawesometable) and the select query fails.

**Comment by tlocke** [comment]
Hi @urkle, thanks for raising this. On balance I think you're right, so I've changed the behaviour here https://github.com/tlocke/pg8000/commit/91b00943814683fc05531da977fd35457b58ce21

**Comment by urkle** [comment]
@tlocke thanks for the quick response!.

Generally IMO I just have quote everything 100% of the time but this change should work fine.

**Comment by tlocke** [comment]
You're right, I think it's best if it just quotes everything, so I've made that change https://github.com/tlocke/pg8000/commit/0a75f1cea65eaab96f09fe5bce342597e907e680

**Comment by urkle** [comment]
closing this issue as the fix is in pg8000 1.31.3+
